plugins/nodes/src/js/components/LeaderGrid.js

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import moment from "moment";
+
+import ConfigurationMap from "#SRC/js/components/ConfigurationMap";
+import ConfigurationMapHeading from "#SRC/js/components/ConfigurationMapHeading";
+import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
+import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
+import ConfigurationMapSection from "#SRC/js/components/ConfigurationMapSection";
+import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
+import Loader from "#SRC/js/components/Loader";
+
+function timeFromNow(time) {
+  if (!time) {
+    return null;
+  }
+
+  return moment(time * 1000).fromNow();
+}
+
+const Loading = () => <Loader size="small" type="ballBeat" />;
+
+const ConfigurationRow = ({ keyValue, title, value }) => {
+  const loadedValue = value || <Loading />;
+
+  return (
+    <ConfigurationMapRow key={keyValue}>
+      <ConfigurationMapLabel>{title}</ConfigurationMapLabel>
+      <ConfigurationMapValue>{loadedValue}</ConfigurationMapValue>
+    </ConfigurationMapRow>
+  );
+};
+
+export default class LeaderGrid extends React.Component {
+  getMesosDetails() {
+    const {
+      hostPort,
+      version,
+      electedTime,
+      startTime,
+      region
+    } = this.props.master;
+
+    return (
+      <ConfigurationMapSection>
+        <ConfigurationRow
+          keyValue="leader"
+          title="IP and Port"
+          value={hostPort}
+        />
+
+        <ConfigurationRow keyValue="region" title="Region" value={region} />
+
+        <ConfigurationRow
+          keyValue="version"
+          title={<FormattedMessage id="COMMON.VERSION" />}
+          value={version}
+        />
+
+        <ConfigurationRow
+          keyValue="version"
+          title={<FormattedMessage id="COMMON.STARTED" />}
+          value={timeFromNow(startTime)}
+        />
+
+        <ConfigurationRow
+          keyValue="elected"
+          title={<FormattedMessage id="COMMON.ELECTED" />}
+          value={timeFromNow(electedTime)}
+        />
+      </ConfigurationMapSection>
+    );
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <ConfigurationMap>
+          <ConfigurationMapHeading className="flush-top">
+            Leader
+          </ConfigurationMapHeading>
+          {this.getMesosDetails()}
```

Comment:
IMHO there is no need extracting this into a function… there is no additional logic 🤔

Reply:
I like your style comment, if there are something people ask me to change, I will go also and fix your comments. Otherwise I will roll with it.